### PR TITLE
ENG-919 Check Details Page

### PR DESCRIPTION
**Pull request by agiron123**
## Describe your changes and why you are making these changes
This PR implements the check details page. I took some liberties on the mocks, but I suspect my mocks were quite out of date.

See mock here:
https://www.figma.com/file/Qz2IiYJ4NAaDHojOf2YoTm/UI-V2-Screens?node-id=874%3A29070

Things I didn't do from the mock:
-Add color tints to the results table. Certainly something we can do, but I think we have some bigger questions to answer.
-Show associated metrics in the related outputs section. (I'm not sure if this is even possible at the moment) Instead, I think we should just show a list of artifacts associated with this Check.
-Render artifact's location. Went with rendering the check's serialized result instead.

Things we should perhaps add that were missing from the mock:
-It would be nice to know what the input to this check was.
-Perhaps we should show the code of the check operator
-Show logs for check if any

I worry that if we add the things above, there wouldn't be much difference at all between the Check Details page and the Metric details page. Are there any other Check specific things that we can do here to give users a better experience?

## Related issue number (if any)
https://linear.app/aqueducthq/issue/ENG-919/implement-check-details-page

## Loom demo (if any)
https://www.loom.com/share/5e5a92120e134bbc959b03306c417a03

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/pages/check/id/index.tsx [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+import AccordionSummary from '@mui/material/AccordionSummary';
+import Alert from '@mui/material/Alert';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link as RouterLink, useParams } from 'react-router-dom';
+import CheckTableItem from '../../../tables/CheckTableItem';
+
+//import PaginatedTable from '../../../../components/tables/PaginatedTable';
```
Comment (agiron123) [inline src/ui/common/src/components/pages/check/id/index.tsx:15]: remove comment.

File: src/ui/common/src/components/pages/check/id/index.tsx [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+    const artifactId = operator?.outputs[0];
+
+    const artifactHistoryWithLoadingStatus = useSelector((state: RootState) =>
+        !!artifactId
+            ? state.artifactResultsReducer.artifacts[artifactId]
+            : undefined
+    );
+
+    console.log('workflowDagResultWithLoadingStatus: ', workflowDagResultWithLoadingStatus);
```
Comment (agiron123) [inline src/ui/common/src/components/pages/check/id/index.tsx:60]: take out logs

File: src/ui/common/src/components/pages/check/id/index.tsx [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+                    >
+                        {artifactResult.name}
+                    </Link>
+                </Box>
+            );
+        }
+
+        return (
+            <Box key={artifactId} display="flex">
+                {/* <Typography variant="body1" marginRight="24px">
```
Comment (agiron123) [inline src/ui/common/src/components/pages/check/id/index.tsx:198]: remove comment

File: src/ui/common/src/components/pages/check/id/index.tsx [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+                            <Typography variant="body1">{operator.description}</Typography>
+                        )}
+                    </Box>
+                </Box>
+
+                <Box width="100%" marginTop="32px">
+                    <Typography variant="h5" marginBottom="8px">Recent Results</Typography>
+                    <CheckHistory historyWithLoadingStatus={artifactHistoryWithLoadingStatus} checkLevel={operator?.spec?.check?.level} />
+                </Box>
+                {/* commenting out metrics for now as we figure out what to do with them */}
```
Comment (agiron123) [inline src/ui/common/src/components/pages/check/id/index.tsx:230]: TODO: Find out if this is going to be something to support. But probably going to end up removing for now and bring back if so.

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 5]
```diff
@@ -323,6 +320,24 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
           </Button>
         </Box>
       );
+    } else if (currentNode.type === NodeType.CheckOp) {
+      // TODO: Show View Check Details
```
Comment (agiron123) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:324]: Remove todo, this has been resolve.

File: src/ui/common/src/components/workflows/artifact/check/history.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import { theme } from '../../../../styles/theme/theme';
+import { Data, DataSchema } from '../../../../utils/data';
+import { isFailed, isInitial, isLoading } from '../../../../utils/shared';
+
+type CheckHistoryProps = {
+    historyWithLoadingStatus?: ArtifactResultsWithLoadingStatus;
+    checkLevel?: string;
+}
+
+// TODO: Bring over data schema from check details page
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/artifact/check/history.tsx:18]: remove comment, this has been resolved.

File: src/ui/common/src/components/workflows/artifact/check/history.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                {historyWithLoadingStatus.status.err}
+            </Alert>
+        );
+    }
+
+    const historicalData: Data = {
+        schema: checkHistorySchema,
+        data: (historyWithLoadingStatus.results?.results ?? []).map(
+            (artifactStatusResult) => {
+                console.log('artifactStatusResult', artifactStatusResult);
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/artifact/check/history.tsx:49]: remove log.

File: src/ui/common/src/components/workflows/artifact/check/history.tsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+                    status: artifactStatusResult.exec_state?.status ?? 'Unknown',
+                    level: checkLevel ? checkLevel : 'undefined',
+                    value: artifactStatusResult.content_serialized,
+                    timestamp: artifactStatusResult.exec_state?.timestamps?.finished_at,
+                };
+            }
+        ),
+    };
+
+    return (
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/artifact/check/history.tsx:60]: TODO: Consider adding in the Plot component similar to the artifact history component that we've based this implementation off of. 

If we do that, we can pass in the artifact type and then render things accordingly i n the history component.

File: src/ui/common/src/components/pages/check/id/index.tsx [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+                {/* <Box width="100%" marginTop="32px">
```
Comment (likawind) [inline src/ui/common/src/components/pages/check/id/index.tsx:231]: I think you can add inputs / outputs section following the pattern in metrics detail page as well (see how `ArtifactSummaryList` get used). Feel free to add as follow ups

File: src/ui/common/src/components/workflows/artifact/check/history.tsx [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+// TODO: Bring over data schema from check details page
+const checkHistorySchema: DataSchema = {
+    fields: [
+        { name: 'status', type: 'varchar' },
+        { name: 'level', type: 'varchar' },
+        { name: 'value', type: 'varchar' },
+        { name: 'timestamp', type: 'varchar' }
+    ],
+    pandas_version: '0.0.1', // Not sure what actual value to put here, just filling in for now :)
```
Comment (likawind) [inline src/ui/common/src/components/workflows/artifact/check/history.tsx:26]: let's use an empty string?

**Review by likawind — APPROVED** [review]
Great! Once we remove console logs and mock data it should be good to merge. A few additional things we could consider in follow-up PRs:
* color code based on success / fail status together with level
* Since level is associated with the operator, not each run, we can render it a 'globally' instead of per every run?

**Pull request merged**